### PR TITLE
Add support for cancel backups

### PR DESCRIPTION
**Pull request by dirkkul**
List backups has been moved back, but I'll leave the code that has already been written in there

### CODE REVIEW

File: weaviate/backup/backup.py [hunk 1]
```diff
@@ -357,6 +365,75 @@ async def get_restore_status(
...
+
+        response = await self._connection.get(
+            path=path, error_msg="Backup list status failed due to connection error."
+        )
+        typed_response = _decode_json_response_list(response, "Backup restore status check")
+        if typed_response is None:
+            raise EmptyResponseException()
+        return [BackupReturn(**entry) for entry in typed_response]
+
+    async def list_backups(self, backend: BackupStorage) -> List[BackupReturn]:
```
Comment (jfrancoa) [inline weaviate/backup/backup.py:422]: This API call needs to be removed as it was disabled in the server side https://github.com/weaviate/weaviate/pull/5797

File: integration_v3/test_backup_v4.py [hunk 2]
```diff
@@ -449,3 +450,37 @@ def test_backup_and_restore_with_collection_and_config_1_23_x(
...
+#     """Create and restore backup without waiting."""
+#     backup_id = _create_backup_id()
+#     if client._connection._weaviate_version.is_lower_than(1, 27, 0):
+#         pytest.skip("List backups is only supported from 1.27.0")
+#
+#     resp = client.backup.create(backup_id=backup_id, backend=BACKEND)
+#     assert resp.status == BackupStatus.STARTED
+#
+#     backups = client.backup.list_backups(backend=BACKEND)
+#     assert backup_id in [b.backup_id for b in backups]
```
Comment (tsmith023) [inline integration_v3/test_backup_v4.py:466]: Is this still true?

File: integration_v3/test_backup_v4.py [hunk 3]
```diff
@@ -449,3 +450,37 @@ def test_backup_and_restore_with_collection_and_config_1_23_x(
...
+
+
+def test_cancel_backup(client: weaviate.WeaviateClient) -> None:
+    """Create and restore backup without waiting."""
+    backup_id = _create_backup_id()
+    if client._connection._weaviate_version.is_lower_than(1, 24, 0):
+        pytest.skip("Cancel backups is only supported from 1.27.0")
```
Comment (tsmith023) [inline integration_v3/test_backup_v4.py:473]: Should `L472` be `is_lower_than(1, 27, 0)`? Or is it available in each currently supported version?

File: weaviate/backup/backup.py [hunk 4]
```diff
@@ -357,6 +366,78 @@ async def get_restore_status(
...
+    async def __cancel_backup(self, backup_id: str, backend: BackupStorage) -> bool:
+        backup_id, backend = _get_and_validate_get_status(
+            backup_id=backup_id,
+            backend=backend,
+        )
+        path = f"/backups/{backend.value}/{backup_id}"
+
+        response = await self._connection.delete(
+            path=path,
+            error_msg="Backup delete failed due to connection error.",
```
Comment (tsmith023) [inline weaviate/backup/backup.py:378]: Despite it being a `DELETE` request, should this say `Backup cancel failed`

File: weaviate/backup/backup.py [hunk 5]
```diff
@@ -357,6 +366,78 @@ async def get_restore_status(
...
+            status_codes=_ExpectedStatusCodes(ok_in=[204, 404], error="delete object"),
+        )
+
+        if response.status_code == 204:
+            return True  # Successfully deleted
+        else:
+            typed_response = _decode_json_response_dict(response, "Backup restore status check")
```
Comment (tsmith023) [inline weaviate/backup/backup.py:385]: Should this be `Backup cancel check`?

File: weaviate/backup/backup.py [hunk 6]
```diff
@@ -357,6 +366,78 @@ async def get_restore_status(
...
+        return await self.__cancel_backup(backup_id, backend)
+
+    async def __list_backups(self, backend: BackupStorage) -> List[BackupReturn]:
+        _, backend = _get_and_validate_get_status(backend=backend, backup_id="dummy")
+        path = f"/backups/{backend.value}"
+
+        response = await self._connection.get(
+            path=path, error_msg="Backup list status failed due to connection error."
+        )
+        typed_response = _decode_json_response_list(response, "Backup restore status check")
```
Comment (tsmith023) [inline weaviate/backup/backup.py:420]: Should this be `Backup list check` or something similar?

File: weaviate/backup/backup.py [hunk 7]
```diff
@@ -357,6 +366,78 @@ async def get_restore_status(
...
+    #     Parameters
+    #     ----------
+    #     backend : BackupStorage
+    #         The backend storage where to create the backup.
+    #
+    #     Returns
+    #     -------
+    #      A list of `BackupStatusReturn` objects that contain the backup restore status responses.
+    #     """
+    #     return await self.__list_backups(backend)
```
Comment (tsmith023) [inline weaviate/backup/backup.py:439]: Is this still true?

File: integration_v3/test_backup_v4.py (hunk 3 above)
Comment (dirkkul) [reply integration_v3/test_backup_v4.py:473]: yep, it was merged into older versions - also the tests work on older versions :)

File: integration_v3/test_backup_v4.py (hunk 2 above)
Comment (dirkkul) [reply integration_v3/test_backup_v4.py:466]: yep: https://github.com/weaviate/weaviate/pull/5918

File: weaviate/backup/backup.py (hunk 7 above)
Comment (dirkkul) [reply weaviate/backup/backup.py:439]: same as above

**Review by tsmith023 — APPROVED** [review]

**Pull request merged**